### PR TITLE
Improve the return of DeletedTreeGC.RunOnce

### PR DESCRIPTION
**Pull request by codingllama**
It now returns the count of successfully deleted trees and any encountered errors.

### CODE REVIEW

File: server/admin/tree_gc_test.go [hunk 1]
```diff
@@ -175,6 +177,176 @@ func TestDeletedTreeGC_RunOnce(t *testing.T) {
...
+
+		listTX := storage.NewMockReadOnlyAdminTX(ctrl)
+		lastTXCall := as.EXPECT().Snapshot(gomock.Any()).Return(listTX, test.snapshotErr)
+		listTX.EXPECT().ListTrees(gomock.Any(), true /* includeDeleted */).AnyTimes().Return(test.trees, test.listErr)
+		listTX.EXPECT().Close().AnyTimes().Return(nil)
+		listTX.EXPECT().Commit().AnyTimes().Return(test.snapshotCommitErr)
+
+		// Sanity check test setup
+		if l1, l2, l3, l4 := len(test.beginErrs), len(test.deleteErrs), len(test.commitErrs), len(test.wantTreeIDs); l1 != l2 || l1 != l3 || l1 != l4 {
+			t.Fatalf("%v: beginErrs, deleteErrs, commitErrs and wantTreeIDs have different lenghts: %v, %v, %v and %v", test.desc, l1, l2, l3, l4)
```
Comment (daviddrysdale) [inline server/admin/tree_gc_test.go]: "lengths" (as Travis points out)

File: server/admin/tree_gc_test.go [hunk 2]
```diff
@@ -175,6 +177,176 @@ func TestDeletedTreeGC_RunOnce(t *testing.T) {
...
+
+			deleteTX := storage.NewMockAdminTX(ctrl)
+			lastTXCall = as.EXPECT().Begin(gomock.Any()).Return(deleteTX, beginErr).After(lastTXCall)
+			deleteTX.EXPECT().HardDeleteTree(gomock.Any(), id).AnyTimes().Return(deleteErr)
+			deleteTX.EXPECT().Close().AnyTimes().Return(nil)
+			deleteTX.EXPECT().Commit().AnyTimes().Return(commitErr)
+		}
+
+		gc := NewDeletedTreeGC(
+			as, deleteThreshold, 1*time.Second /* minRunInterval */, nil /* mf */)
```
Comment (daviddrysdale) [inline server/admin/tree_gc_test.go]: (nit) line break seems unnecessary here

File: server/admin/tree_gc_test.go [hunk 3]
```diff
@@ -175,6 +177,176 @@ func TestDeletedTreeGC_RunOnce(t *testing.T) {
...
+		as := storage.NewMockAdminStorage(ctrl)
+
+		listTX := storage.NewMockReadOnlyAdminTX(ctrl)
+		lastTXCall := as.EXPECT().Snapshot(gomock.Any()).Return(listTX, test.snapshotErr)
+		listTX.EXPECT().ListTrees(gomock.Any(), true /* includeDeleted */).AnyTimes().Return(test.trees, test.listErr)
+		listTX.EXPECT().Close().AnyTimes().Return(nil)
+		listTX.EXPECT().Commit().AnyTimes().Return(test.snapshotCommitErr)
+
+		// Sanity check test setup
+		if l1, l2, l3, l4 := len(test.beginErrs), len(test.deleteErrs), len(test.commitErrs), len(test.wantTreeIDs); l1 != l2 || l1 != l3 || l1 != l4 {
```
Comment (daviddrysdale) [inline server/admin/tree_gc_test.go]: Not worth fixing for a test, but this self-consistency check hints you might be better off having a `[]testOp` with `testOp` being a struct holding a begin/delete/commit error and a treeID, so it's impossible for a maintainer to create a new invalid testcase.

File: server/admin/tree_gc_test.go [hunk 4]
```diff
@@ -175,6 +177,176 @@ func TestDeletedTreeGC_RunOnce(t *testing.T) {
...
+		}
+
+		failed := false
+		for _, want := range test.wantErrs {
+			if !strings.Contains(err.Error(), want) {
+				if !failed {
+					t.Errorf("%v: RunOnce() returned err = %v (see following errors)", test.desc, err)
+					failed = true
+				}
+				t.Errorf("%v: RunOnce(): err doesn't contain %q", test.desc, want)
```
Comment (daviddrysdale) [inline server/admin/tree_gc_test.go]: You could just include `err` in this log and drop the `failed` stuff.  Or not, up to you.

File: server/admin/tree_gc_test.go (hunk 3 above)
Comment (codingllama) [reply server/admin/tree_gc_test.go]: Good call. I went ahead and did this, as it makes it a lot easier to read the expected hard-delete flow.

File: server/admin/tree_gc_test.go (hunk 1 above)
Comment (codingllama) [reply server/admin/tree_gc_test.go]: Ack (line removed).

File: server/admin/tree_gc_test.go (hunk 4 above)
Comment (codingllama) [reply server/admin/tree_gc_test.go]: Done. I thought it would look terrible, but it isn't that bad tbh.

File: server/admin/tree_gc_test.go (hunk 2 above)
Comment (codingllama) [reply server/admin/tree_gc_test.go]: Done.

**Comment by codingllama** [comment]
Comments addressed.

**Pull request merged**